File: accepted/205-isomorphic-strings.py
# https://leetcode.com/problems/isomorphic-strings/
class Solution:
    # @param {string} s
    # @param {string} t
    # @return {boolean}
    def isIsomorphic(self, s, t):
        l = len(s)
        if l != len(t):
            return False
        
        map1 = {}
        map2 = {}
        for i in xrange(l):
            if s[i] in map1 and map1[s[i]] != t[i]:
                return False
            elif t[i] in map2 and map2[t[i]] != s[i]:
                return False
            map1[s[i]] = t[i]
            map2[t[i]] = s[i]
            
        return True

File: accepted/205-isomorphic-strings.c
/* https://leetcode.com/problems/isomorphic-strings/ */
bool isIsomorphic(char* s, char* t) {
    char map1[128] = {0};
    char map2[128] = {0};
    
    int i = 0;
    while ('\0' != s[i]) {
        if (map1[s[i]] != 0 && map1[s[i]] != t[i])
            return false;
        if (map2[t[i]] != 0 && map2[t[i]] != s[i])
            return false;
        
        map1[s[i]] = t[i];
        map2[t[i]] = s[i];
        
        i++;
    }
    
    return true;
}
